src/public/services/FormService.ts

```diff
@@ -0,0 +1,134 @@
+import axios from 'axios'
+
+import { FormUpdateParams } from 'src/app/modules/form/admin-form/admin-form.types'
+import { PublicFormViewDto } from 'src/app/modules/form/public-form/public-form.types'
+import {
+  FormMetaView,
+  IFormSchema,
+  IPopulatedForm,
+  PublicForm,
+} from 'src/types'
+import { FormFieldDto } from 'src/types/api'
+
+// endpoints exported for testing
+export const ADMIN_FORM_ENDPOINT = '/api/v3/admin/forms'
+export const PUBLIC_FORM_ENDPOINT = '/api/v3/forms'
+
+export const queryForm = async (): Promise<FormMetaView[]> => {
+  const instance = axios.create()
+  return instance
+    .get<FormMetaView[]>(`${ADMIN_FORM_ENDPOINT}`, {
+      headers: { 'If-Modified-Since': '0' },
+    })
+    .then(({ data }) => data)
+}
+
+export const getAdminForm = async (
+  formId: string,
+  accessMode = 'adminform',
+): Promise<{ form: IPopulatedForm }> => {
+  const instance = axios.create()
+  // disable IE ajax request caching (so new forms show on dashboard)
+  return instance
+    .get<{ form: IPopulatedForm }>(`/${formId}/${accessMode}`, {
+      headers: { 'If-Modified-Since': '0' },
+    })
+    .then(({ data }) => data)
+}
+
+export const getPublicForm = async (
+  formId: string,
+): Promise<PublicFormViewDto> => {
+  const instance = axios.create()
+  // disable IE ajax request caching (so new forms show on dashboard)
+  return instance
+    .get<PublicFormViewDto>(`${PUBLIC_FORM_ENDPOINT}/${formId}`, {
+      headers: { 'If-Modified-Since': '0' },
+    })
+    .then(({ data }) => data)
+}
+
+export const updateForm = async (
+  formId: string,
+  update: { form: FormUpdateParams },
+  accessMode = 'adminform',
+): Promise<IPopulatedForm> => {
+  const instance = axios.create()
+  return instance
+    .put<IPopulatedForm>(`${formId}/${accessMode}`, update)
+    .then(({ data }) => data)
+}
+
+export const saveForm = async (
+  formId: string,
+  formToSave: FormFieldDto[],
```

Comment:
not sure this typing is correct. if you refer to the backend /duplicate endpoint, the type should be `DuplicateFormBody`, which means the argument should be named similarly too